default/python/AI/PlanetUtilsAI.py

```diff
@@ -180,10 +199,47 @@ def get_planet_opinion(feature: str) -> Opinion:
     return _calculate_get_planet_opinions().get(feature, default)
 
 
+@cache_for_current_turn
+def _planned_species() -> Mapping[PlanetId, str]:
+    universe = fo.getUniverse()
+    production_queue = fo.getEmpire().productionQueue
+    planned_species = {}
+    building_prefix = "BLD_COL_"
+    species_prefix = "SP_"
+    colonisation_plans = get_aistate().colonisablePlanetIDs
+    for element in production_queue:
+        if element.name.startswith(building_prefix):
+            planned_species[element.locationID] = species_prefix + element.name[len(building_prefix) :]
+    for pid in get_owned_planets_by_empire():
+        planet = universe.getPlanet(pid)
+        for building in map(universe.getBuilding, planet.buildingIDs):
+            if building.name.startswith(building_prefix):
+                planned_species[pid] = species_prefix + building.name[len(building_prefix) :]
+                continue
+        # Without checking the colonisation plans, the AI may start building a colony and buildings
+        # the future species wouldn't like in the same turn.
+        plan = colonisation_plans.get(pid)
+        if plan:
+            planned_species[pid] = plan[1]
+    debug(f"Planned species: {planned_species}")
+    return planned_species
+
+
+def _planet_species(pid: PlanetId) -> Optional[fo.species]:
+    universe = fo.getUniverse()
+    planet = universe.getPlanet(pid)
+    species_name = planet.speciesName
+    if not species_name:
+        species_name = _planned_species().get(pid)
+    if species_name:
+        return fo.getSpecies(species_name)
```

Comment:
This function returns a value, so let's explicitly return None.

```suggestion
    if species_name:
        return fo.getSpecies(species_name)
    else:
        return None
```